algorithms/src/Kokkos_Sort.hpp

```diff
@@ -533,6 +533,7 @@ bool try_std_sort(ViewType view) {
   possible = possible && (ViewType::Rank == 1);
   possible = possible && (stride[0] == 1);
   if (possible) {
+    exec.fence("Kokkos::sort: Fence before sorting on the host");
```

Comment:
Although I don't necessarily object to that change, it needs to be highlighted not just committed along with addressing another unrelated review comment.

Reply:
I split the commit.